packages/schemas/alterations/next-1665300135-sign-in-sign-up.ts

```diff
@@ -0,0 +1,174 @@
+    [SignInMethodKey.Sms]: SignInMethodState;
+    [SignInMethodKey.Social]: SignInMethodState;
+  };
+};
+
+enum SignUpIdentifier {
+  Email = 'email',
+  Phone = 'phone',
+  Username = 'username',
+  EmailOrPhone = 'emailOrPhone',
+  None = 'none',
+}
+
+enum SignInIdentifier {
+  Email = 'email',
+  Phone = 'phone',
+  Username = 'username',
+}
+
+type SignIn = {
+  methods: Array<{
+    identifier: SignInIdentifier;
+    password: boolean;
+    verificationCode: boolean;
+    isPasswordPrimary: boolean;
+  }>;
+};
+
+type SignUp = {
+  identifier: SignUpIdentifier;
+  password: boolean;
+  verify: boolean;
+};
+
+const parseSignInMethodToSignInIdentifier = (
+  method: SignInMethodKey
+): SignInIdentifier | undefined => {
+  if (method === SignInMethodKey.Username) {
+    return SignInIdentifier.Username;
+  }
+
+  if (method === SignInMethodKey.Email) {
+    return SignInIdentifier.Email;
+  }
+
+  if (method === SignInMethodKey.Sms) {
+    return SignInIdentifier.Phone;
+  }
+};
+
+const parseSignInMethodToSignUpIdentifier = (method?: SignInMethodKey): SignUpIdentifier => {
+  if (method === SignInMethodKey.Username) {
+    return SignUpIdentifier.Username;
+  }
+
+  if (method === SignInMethodKey.Email) {
+    return SignUpIdentifier.Email;
+  }
+
+  if (method === SignInMethodKey.Sms) {
+    return SignUpIdentifier.Phone;
+  }
+
+  return SignUpIdentifier.None;
+};
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      alter table sign_in_experiences add column sign_in jsonb
+    `);
+    await pool.query(sql`
+      alter table sign_in_experiences add column sign_up jsonb
+    `);
+    await pool.query(sql`
+      alter table sign_in_experiences add column forgot_password boolean not null default false
+    `);
+
+    const id = 'default';
+    const data = await pool.maybeOne<SignInExperience>(
+      sql`select * from sign_in_experiences where id = ${id}`
+    );
+
+    /* eslint-disable @silverhand/fp/no-mutating-methods */
+    if (data) {
+      const { sign_in_methods: signInMethods } = data;
+      const methodKeys = Object.values(SignInMethodKey);
+      const primaryMethod = methodKeys.find(
+        (key) => signInMethods[key] === SignInMethodState.Primary
+      );
+      const secondaryMethods = methodKeys.filter(
+        (key) => signInMethods[key] === SignInMethodState.Secondary
+      );
+      const signIn: SignIn = {
+        methods: [],
+      };
+
+      const methods = [primaryMethod, ...secondaryMethods];
+
+      for (const method of methods) {
+        if (!method) {
+          continue;
+        }
+
+        const identifier = parseSignInMethodToSignInIdentifier(method);
+
+        if (identifier) {
+          signIn.methods.push({
+            identifier,
+            password: identifier === SignInIdentifier.Username,
+            verificationCode: identifier !== SignInIdentifier.Username,
+            isPasswordPrimary: false,
```

Comment:
nit: isPasswordPrimary = identifier === SignInIdentifier.Username

Reply:
The old version don't support email + password sign in, so I'll keep "false"